### ISSUE TITLE
Recovering/reconnecting to session + `xpra upgrade`

### ISSUE THREAD

**Issue by stdedos** [Xpra-org/xpra#3175]
Somewhat related to https://github.com/Xpra-org/xpra/issues/3088:

`xpra upgrade`ing server to test https://github.com/Xpra-org/xpra/issues/3139 leads to the following situation:

- xpra server is attempted to be replaced, but it fails.
- x2 ssh with `xpra attach` fails with:
  ```
  2021-06-21 13:29:25,410 Error: failed to receive anything, not an xpra server?
  2021-06-21 13:29:25,422   could also be the wrong protocol, username, password or port
  2021-06-21 13:29:25,425   or the session was not found
  2021-06-21 13:29:25,427 Connection lost
  2021-06-21 13:29:25,459 remote SSH stderr:
  2021-06-21 13:29:25,463  server socket for display :2 not found
  2021-06-21 13:29:25,464   waiting up to 20 seconds
  2021-06-21 13:29:25,465  xpra initialization error:
  2021-06-21 13:29:25,467   cannot find live server for display :2
    
  was: "Xpra-Python3-x86_64_4.3-r29325\xpra_cmd" start ssh://user@ip/2 --ssh="plink -ssh -agent" --title="@title@ on @@/@server-display@" --modal-windows=no --headerbar=off --use-display=yes --mousewheel=invert-y --key-shortcut=#+R:refresh_all_windows --opengl=no --bandwidth-limit=6Mbps
  ```
  (and even waiting for the xpra list full 20s timeout does not kill the displays)

Resolution: Connect with an adjacent xpra display, and run `xpra recover`.

**Comment by totaam** [comment]
I believe that this is now fixed by dc05eabb222ad5c5bc46b0d9486cb6c6b3c4a0e6

Other related improvements in 94cc4502cdb4a017f9a6a2d1e2858eb3dac83aab, 06bd62a008bc6d8de988fa6793e7209319552fdd, 641fcef2d79f411b840c7fbf6da046be4032d3f8

**Comment by stdedos** [comment]
Yeap, xpra upgrade to r29386 was flawless
